overlord/hookstate/hookmgr.go

```diff
@@ -288,6 +310,15 @@ func runHookAndWait(snapName string, revision snap.Revision, hookName, hookConte
 		return nil, err
 	}
 
+	// add timeout handling
```

Comment:
Any reason not to use time.After here?  Should make much of the boilerplate below go away.